planners/astar/planner.c

```diff
@@ -93,14 +94,19 @@ static int trajectory_get_route_update(ctrl_t* ctrl, const pose_t* robot_pose,
             }
             /* Update current path targeted position in case it has changed */
             current_path_pos = path_get_current_path_pos(path);
+
             /* As current path pose could have changed, avoidance graph needs
-             * to be recomputed */
+                * to be recomputed */
+            avoidance_update = TRUE;
+        }
+        /* If in calibration mode and new pose is selected */
+        else if ((!allow_change_path_pose) && (ctrl_is_pose_intermediate(ctrl))) {
```

Comment:
if we have reach an intermediate position it should be `avoidance_index++;`, not `avoidance_update = TRUE;`.

Thats why I deleted it in a previous commit. What do you think ?

Reply:
I've just restored the former code that was working, but I don't clearly understand the planner behavior...